invenio_communities/communities/schema.py

```diff
@@ -265,14 +265,25 @@ class CommunitySchema(BaseCommunitySchema):
     """Community schema."""
 
     parent = NestedAttribute(CommunityParentSchema, dump_only=True)
-
+    # When loading the parent community we don't require any fields as we will only use the ID
+    id = fields.String(dump_only=False)
+    metadata = NestedAttribute(CommunityMetadataSchema, required=False) 
+    access = NestedAttribute(CommunityAccessSchema, required=False)
+    
     @post_dump
     def post_dump(self, data, many, **kwargs):
         """Hide parent field if it's not present."""
         data = super().post_dump(data, many, **kwargs)
         if data.get("parent") is None:
             data.pop("parent", None)
         return data
+    
+    @post_load
+    def filter_parent_id(self, in_data, **kwargs):
+        """Simply keep the parent id."""
+        if in_data.get("parent"):
+            in_data["parent"] = dict(id=in_data.get("parent", {}).get("id"))
```

Comment:
minor: there are two input cases we have to handle:

- `{"parent": {"id": "<some_id>"}` -  we filter and just keep the `parent.id` key
- `{"parent": None}` - We keep the data as is so that we can "unset" the community

Reply:
I think that @yashlamba is working on the service method that allows to set/update the parent, maybe you can improve it as you can test it